### PR TITLE
Sarinali/cfd 49 schedule frontend page

### PR DESCRIPTION
**Pull request by sarinali**
### Description
Made the changes on NextJS frontend

### Reason for Change
CFD49

- [x] New feature (non-breaking change which adds functionality)

### Test
in `apps/nextjs/src/pages/index.tsx` add `<Activities></Activities>`, dont forget the import statement: `import Activities from "~/components/activity/activities";`

https://github.com/uoftblueprint/centre-for-dreams/assets/74844953/409fa91c-d21a-41b1-a714-c7846fba502f

### CODE REVIEW

**Comment by BakerWJ** [comment]
can u revert all of the changes in the expo dir?

**Comment by sarinali** [comment]
> can u revert all of the changes in the expo dir?

yahh doing rn

File: apps/nextjs/package-lock.json
Comment (BakerWJ) [inline apps/nextjs/package-lock.json:1]: remove

File: apps/expo/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "react-dom": "18.2.0",
     "react-native": "0.72.6",
     "react-native-gesture-handler": "~2.12.0",
+    "react-native-modal-datetime-picker": "^17.1.0",
```
Comment (BakerWJ) [inline apps/expo/package.json:38]: remove

File: apps/nextjs/src/components/activity/create_activity.tsx
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/create_activity.tsx:1]: maybe rename to ActivityForm

File: apps/nextjs/src/components/activity/create_activity.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import "react-datepicker/dist/react-datepicker.css";
+
+export default function CreateActivity() {
+  const currentDateTime: Date = new Date();
+  const [selectedDate, setSelectedDate] = useState<Date | null>(
+    currentDateTime,
+  );
+  const [selectedStartDate, setselectedStartDate] = useState<Date | null>(
+    currentDateTime,
+  );
```
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/create_activity.tsx:15]: does this need a null type?

File: apps/nextjs/src/components/activity/create_activity.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      name: "",
+      duration: "",
+      leader: "",
+      location: "",
+      selectedDate: currentDateTime,
+      startDate: currentDateTime,
+    });
+
+    alert("Submitted!");
+  };
```
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/create_activity.tsx:67]: for the form can you use react-hook-form? you can see an example at https://github.com/uoftblueprint/centre-for-dreams/blob/main/apps/nextjs/src/pages/createpost.tsx

File: apps/nextjs/src/components/activity/create_activity.tsx (hunk 2 above)
Comment (sarinali) [reply apps/nextjs/src/components/activity/create_activity.tsx]: removed

File: apps/nextjs/src/components/activity/create_activity.tsx
Comment (sarinali) [reply apps/nextjs/src/components/activity/create_activity.tsx:1]: done

File: apps/expo/package.json (hunk 1 above)
Comment (sarinali) [reply apps/expo/package.json:38]: ok

File: apps/nextjs/package-lock.json
Comment (BakerWJ) [inline apps/nextjs/package-lock.json:1]: delete

File: apps/nextjs/src/components/activity/activities.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+import CreateActivity from "./activityform";
+import ViewActivities from "./viewactivites";
+
+const Activities = () => {
+  return (
+    <div className="h-full w-full p-4">
+      <div className="font-medium">Create Activity</div>
+      <CreateActivity></CreateActivity>
```
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/activities.tsx:10]:
```suggestion
      <CreateActivity />
```

File: apps/nextjs/src/components/activity/activities.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+      <div className="font-medium">View Activites</div>
+      <ViewActivities></ViewActivities>
```
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/activities.tsx:12]:
```suggestion
      <ViewActivities />
```

File: apps/nextjs/src/pages/index.tsx [hunk 6]
```diff
@@ -10,6 +10,8 @@ import RyanLi from "~/components/developers/ryanli";
 import SarinaLi from "~/components/developers/sarinali";
 import { api } from "~/utils/api";
 
+// import CreateActivity1 from "~/components/activity/create_activity1";
+
```
Comment (BakerWJ) [inline apps/nextjs/src/pages/index.tsx:14]: delete

File: pnpm-lock.yaml [hunk 7]
```diff
@@ -83,6 +83,9 @@ importers:
       react-native-gesture-handler:
         specifier: ~2.12.0
         version: 2.12.1(react-native@0.72.6)(react@18.2.0)
+      react-native-modal-datetime-picker:
+        specifier: ^17.1.0
+        version: 17.1.0(@react-native-community/datetimepicker@7.6.1)(react-native@0.72.6)
```
Comment (BakerWJ) [inline pnpm-lock.yaml:88]: looks like lockfile isn't up to date?

File: apps/nextjs/src/components/activity/create_activity.tsx (hunk 3 above)
Comment (sarinali) [reply apps/nextjs/src/components/activity/create_activity.tsx]: fixed

File: apps/nextjs/src/components/activity/activityform.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+import { api } from "~/utils/api";
+
+interface ActivityData {
+  name: string;
+  duration: string;
+  leader: string;
+  location: string;
+  selectedDate: Date;
+  startDate: Date;
+}
```
Comment (BakerWJ) [inline apps/nextjs/src/components/activity/activityform.tsx:14]: you can use https://github.com/uoftblueprint/centre-for-dreams/blob/main/apps/nextjs/src/utils/api.ts#L69 for the type that the endpoint accepts

File: pnpm-lock.yaml (hunk 7 above)
Comment (BakerWJ) [reply pnpm-lock.yaml:88]: you can fix by running `pnpm i` in the root directory

File: apps/nextjs/src/components/activity/activityform.tsx (hunk 8 above)
Comment (sarinali) [reply apps/nextjs/src/components/activity/activityform.tsx:14]: i dont know how to do that i asked chatgpt and im still confused, do i define my own type inside the api file and then infer it in the activity form file? 😭

Comment (BakerWJ) [reply apps/nextjs/src/components/activity/activityform.tsx:14]: image


like this -- this is similar to what you can do for the other component PRs (except those are using the outputs)

**Comment by BakerWJ** [comment]
can u fix the merge conflicts so that checks can run?

File: pnpm-lock.yaml
Comment (BakerWJ) [inline pnpm-lock.yaml:1]: revert changes in this file

**Review by BakerWJ — COMMENTED** [review]
is there not an actual page where these components are used?

File: pnpm-lock.yaml
Comment (BakerWJ) [reply pnpm-lock.yaml:1]: pls fix @sarinali why is it showing changes im confused 😢

Comment (sarinali) [reply pnpm-lock.yaml:1]: i just took the pnpm lock from main i hope thats ok

**Review by BakerWJ — APPROVED** [review]

**Pull request merged**